src/plugins/ucx/ucx_backend.cpp

```diff
@@ -629,14 +666,16 @@ nixl_status_t nixlUcxEngine::disconnect(const std::string &remote_agent) {
         //agent names should never be long enough to need RNDV
         flags |= UCP_AM_SEND_FLAG_EAGER;
 
-        ret = uw->sendAm(conn.ep, DISCONNECT,
-                        &hdr, sizeof(struct nixl_ucx_am_hdr),
-                        (void*) localAgent.data(), localAgent.size(),
-                        flags, req);
-
-        //don't care
-        if(ret == NIXL_IN_PROG){
-            uw->reqRelease(req);
+        nixl_status_t ret = NIXL_SUCCESS;
+        for (size_t i = 0; i < uws.size(); i++) {
+            nixlUcxReq req;
+            ret = getWorker(i)->sendAm(conn.getEp(i), DISCONNECT,
+                                 &hdr, sizeof(struct nixl_ucx_am_hdr),
+                                 (void*) localAgent.data(), localAgent.size(),
+                                 flags, req);
+            //don't care
```

Comment:
maybe unite common code with nixlUcxEngine::connect?

Reply:
i guess it will be a function the gets the active message id (CONN_CHECK/DISCONNECT) and sends it on all workers and returns a vector of requests. here we can discard it.

Reply:
And what do you do in case of an error on Nth sendAm() call? Return some ugly pair<errcode, req_vector> so caller could manually call uw->test() or uw->reqRelease() on all the previous successful requests?